jans-cedarling/cedarling/src/init/cedar_schema.rs

```diff
@@ -0,0 +1,81 @@
+/*
+ * This software is available under the Apache-2.0 license.
+ * See https://www.apache.org/licenses/LICENSE-2.0.txt for full text.
+ *
+ * Copyright (c) 2024, Gluu, Inc.
+ */
+
+use base64::prelude::*;
+
+const MSG_UNABLE_DECODE_SCHEMA_BASE64: &str = "unable to decode cedar policy schema base64";
```

Comment:
These errors should be handled using the `CedarlingApplicationError ` enum.

Reply:
We use it in the function `parse_cedar_schema`. It is wrapper to seamless deserialize `cedar_policy::Schema` from out JSON policy store.

Here how it is used in the macros:
```
#[derive(Debug, serde::Deserialize)]
pub struct PolicyStore {
    #[serde(deserialize_with = "parse_cedar_schema")]
    pub schema: cedar_policy::Schema,
}
```

In function, we can't return our custom type. Because we return generic of `serde::Deserializer`
```
pub(crate) fn parse_cedar_schema<'de, D>(deserializer: D) -> Result<cedar_policy::Schema, D::Error>
where
    D: serde::Deserializer<'de>,
{
...
```

Here is [link](https://serde.rs/error-handling.html) with documentation about custom error handling in the `serde`.
So to return error from custom `Deserializer` we need to use implementation
```
impl de::Error for Error {
    fn custom<T: Display>(msg: T) -> Self {
        Error::Message(msg.to_string())
    }
}
```
Where error is casting to the string.

And actually it was made with
```
        serde::de::Error::custom(format!("{MSG_UNABLE_DECODE_SCHEMA_BASE64}: {}", err,))
```

and it was cowered by:
```
pub enum ErrorLoadPolicyStore {
    #[error("{0}")]
    JsonParce(#[from] serde_json::Error),
    ....
    ```

Reply:
I was just saying that we should maintain uniformity of code everywhere. If we are taking the error messages as enum items somewhere then let's make it a practice everywhere. and we can easily convert an enum to a string. ref: `https://stackoverflow.com/questions/32710187/how-do-i-get-an-enum-as-a-string`.

Reply:
Otherwise, we need to create struct with RAW data and manually call a function to decode each field.
Which approach do you think is better?

Reply:
as discussed, we can do something like the below to solve `serde` error handling
```
fn custom_deserialization_error(err: base64::DecodeError, errorEnum: ErrorLoadPolicyStore) -> de::value::Error {
    let message = errorEnum.to_string(); // Using the enum variant
    serde::de::Error::custom(format!("{}: {}", message, err))
}
```

Reply:
We discussed and decided to stay with const messages approach